docs/source/dev_guide/algorithms/kmeans/computation-distributed.rst

```diff
@@ -0,0 +1,220 @@
+       with 32-bit integer assignments of cluster indices to feature vectors in the input data on the local node.
+       
+       .. note::
+            By default, this result is an object of the ``HomogenNumericTable`` class,
+            but you can define this result as an object of any class derived from ``NumericTable``
+            except ``PackedTriangularMatrix``, ``PackedSymmetricMatrix``, and ``CSRNumericTable``.
+
+.. _kmeans_computation_step_2:
+
+Step 2 - on Master Node
++++++++++++++++++++++++
+
+.. image:: images/kmeans-distributed-computation-step-2.png
+    :width: 1000
+
+In this step, the K-Means clustering algorithm accepts the input from each local node described below.
+Pass the ``Input ID`` as a parameter to the methods that provide input for your algorithm.
+For more details, see :ref:`algorithms`.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 60
+   :align: left
+
+   * - Input ID
+     - Input
+   * - ``partialResuts``
+     - A collection that contains results computed in :ref:`Step 1 <kmeans_computation_step_1>` on local nodes.
+
+In this step, the K-Means clustering algorithm calculates the results described below.
+Pass the ``Result ID`` as a parameter to the methods that access the results of your algorithm.
+For more details, see :ref:`algorithms`.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 60
+   :align: left
+
+   * - Result ID
+     - Result
+   * - ``centroids``
+     - Pointer to the :math:`\mathrm{nClusters} \times p` numeric table with centroids. 
+
+       .. note::
+          
+          By default, this result is an object of the ``HomogenNumericTable`` class,
+          but you can define the result as an object of any class derived from ``NumericTable``
+          except ``PackedTriangularMatrix``, ``PackedSymmetricMatrix``, and ``CSRNumericTable``.
+   * - ``objectiveFunction``
+     - Pointer to the :math:`1 \times 1` numeric table that contains the value of the objective function.
+       
+       .. note::
+            By default, this result is an object of the ``HomogenNumericTable`` class,
+            but you can define this result as an object of any class derived from ``NumericTable`` except ``CSRNumericTable``.
+
+.. important::
+    The algorithm computes assignments using input centroids.
+    Therefore, to compute assignments using final computed centroids, after the last call to ``Step2compute()`` method on the master node,
+    on each local node set assignFlag to true and do one additional call to ``Step1compute()`` and ``finalizeCompute()`` methods.
+    Always set assignFlag to true and call ``finalizeCompute()`` to obtain assignments in each step.
+
+.. note::
+    To compute assignments using original ``inputCentroids`` on the given node,
+    you can use K-Means clustering algorithm in the batch processing mode with the subset of the data available on this node.
+    See :ref:`kmeans_computation_batch` for more details.
```

Comment:
Does it make sense to mention that the algorithm itself doesn't provide any strategy to handle empty clusters?